### PR TITLE
separate home and remote decimals when deploying erc20 remote

### PR DESCRIPTION
**Pull request by felipemadero**

### CODE REVIEW

File: cmd/interchaincmd/tokentransferrercmd/deploy.go [hunk 1]
```diff
@@ -578,12 +577,11 @@ func CallDeploy(_ []string, flags DeployFlags) error {
 		remoteSupply  *big.Int
 	)
 
-	tokenRemoteDecimals := tokenHomeDecimals
-	if flags.remoteFlags.tokenDecimals != 0 {
-		tokenRemoteDecimals = flags.remoteFlags.tokenDecimals
-	}
-
 	if !flags.remoteFlags.native {
+		tokenRemoteDecimals := tokenHomeDecimals
```
Comment (michaelkaplan13) [inline cmd/interchaincmd/tokentransferrercmd/deploy.go:581]: So for ERC20TokenRemote, we default the `tokenDecimals` to be the same as the `tokenHomeDecimals`, but allow this to be overridden by a user provided flag? This seems perfectly reasonable to me, could we just add a comment here documenting the behavior?

**Review by arturrez — APPROVED** [review]

**Review by michaelkaplan13 — APPROVED** [review]
Makes sense to me.

Just want to confirm, it is always the case that either:
- `flags.homeFlags.homeAddress != ""`
- `flags.homeFlags.erc20Address != ""`
- `flags.homeFlags.native`

Otherwise, it appears the `tokenHomeSymbol`, `tokenHomeName`, and `tokenHomeDecimals` are never properly set.

**Comment by felipemadero** [comment]
> Makes sense to me.
> 
> Just want to confirm, it is always the case that either:
> 
> * `flags.homeFlags.homeAddress != ""`
> * `flags.homeFlags.erc20Address != ""`
> * `flags.homeFlags.native`
> 
> Otherwise, it appears the `tokenHomeSymbol`, `tokenHomeName`, and `tokenHomeDecimals` are never properly set.

Yes, we check for this on https://github.com/ava-labs/avalanche-cli/blob/c8cfe5ebfdf9132021a804463d211a9dde94da56/cmd/interchaincmd/tokentransferrercmd/deploy.go#L129

**Comment by michaelkaplan13** [comment]
> > Makes sense to me.
> > Just want to confirm, it is always the case that either:
> > 
> > * `flags.homeFlags.homeAddress != ""`
> > * `flags.homeFlags.erc20Address != ""`
> > * `flags.homeFlags.native`
> > 
> > Otherwise, it appears the `tokenHomeSymbol`, `tokenHomeName`, and `tokenHomeDecimals` are never properly set.
> 
> Yes, we check for this on
> 
> https://github.com/ava-labs/avalanche-cli/blob/c8cfe5ebfdf9132021a804463d211a9dde94da56/cmd/interchaincmd/tokentransferrercmd/deploy.go#L129

It looks like that function checks that no more than 1 of the conditions is true, but doesn't ensure that at least 1 of them are true. It may not be a big issue either way though.

**Pull request merged**